zipkin-server/src/main/java/zipkin2/server/internal/ZipkinServerConfiguration.java

```diff
@@ -65,23 +73,21 @@ public void addViewControllers(ViewControllerRegistry registry) {
     registry.addRedirectViewController("/info", "/actuator/info");
   }
 
-  @Bean
-  public UndertowServletWebServerFactory embeddedServletContainerFactory(
-      @Value("${zipkin.query.allowed-origins:*}") String allowedOrigins) {
-    UndertowServletWebServerFactory factory = new UndertowServletWebServerFactory();
-    CorsHandler cors = new CorsHandler(allowedOrigins);
-    if (httpCollector != null) {
-      factory.addDeploymentInfoCustomizers(
-          info -> info.addInitialHandlerChainWrapper(httpCollector));
-    }
-    factory.addDeploymentInfoCustomizers(info -> info.addInitialHandlerChainWrapper(cors));
-    if (httpTracingCustomizer != null) {
-      factory.addDeploymentInfoCustomizers(httpTracingCustomizer);
-    }
-    if (httpRequestDurationCustomizer != null) {
-      factory.addDeploymentInfoCustomizers(httpRequestDurationCustomizer);
-    }
-    return factory;
+  @Bean ArmeriaServerConfigurator corsConfigurator(
+    @Value("${zipkin.query.allowed-origins:*}") String allowedOrigins) {
+    CorsServiceBuilder corsBuilder = allowedOrigins.equals("*") ? CorsServiceBuilder.forAnyOrigin()
+      : CorsServiceBuilder.forOrigins(allowedOrigins.split(","));
+
+    Function<Service<HttpRequest, HttpResponse>, CorsService>
+      corsDecorator = corsBuilder.allowRequestMethods(HttpMethod.GET).newDecorator();
+
+    return server -> server
```

Comment:
Actually it's enough that you just specify `return sb -> sb.decorator(corsDecorator);`. But I guess we have a problem CORS with annotated services. Let me check it, then let you know.

Reply:
Hi @hyangtack - as that issue has been fixed, I tried deleting the below preflight hack but it still didn't work. Is there something else that needs to be done too?

Reply:
Really? I will take a look again!

Reply:
follow-up PRs welcome. I mainly merged this to get @anuraaga out of the work pit of rebasing massive shifts. looks good enough to move forward with, but yeah any follow-ups very welcome!

Reply:
@hyangtack Thanks! By the way I should have mentioned that by "didn't work", I meant the integration test didn't succeed. I didn't dig at all so there is also the chance the test itself needs an update and not Armeria (I don't know cors very well and it's not something I want to add to my life ;) )